cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -167,13 +181,111 @@ void deleteOldFoldersIfExceedLimit(File baseDirectory) throws IOException {
     }
   }
 
+  /**
+   * Deletes select folder
+   */
+  public void deleteBundle(String uuid) throws IOException, NotFoundException {
+    File uuidFile = getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+    DirUtils.deleteDirectoryContents(uuidFile);
+  }
+
+  /**
+   * Get single support bundle overall status
+   */
+  public SupportBundleOperationStatus getBundle(String uuid) throws IOException, NotFoundException {
+    File uuidFile = getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+
+    File statusFile = new File(uuidFile, "status.json");
+    if (!statusFile.exists()) {
+      throw new NotFoundException(String.format("The status for this bundle: %s is not found.", uuid));
+    }
+    SupportBundleStatus bundleStatus = getBundleStatus(uuidFile);
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet = bundleStatus.getTasks();
+    return new SupportBundleOperationStatus(uuidFile.getName(), bundleStatus.getStatus(), supportBundleTaskStatusSet);
+  }
+
+  /**
+   * Archive support bundle into zip file
+   */
+  public String createBundleZipByRequest(String uuid, Path tmpPath, SupportBundleRequestFileList bundleRequestFileList)
+    throws IOException, NotFoundException, NoSuchAlgorithmException {
+    File uuidFile = getUUIDFile(uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("This bundle id %s is not existed", uuid));
+    }
+
+    MessageDigest digest = MessageDigest.getInstance("SHA-256");
+
+    try (ZipOutputStream zipOut = new ZipOutputStream(
+      new DigestOutputStream(Files.newOutputStream(tmpPath, StandardOpenOption.TRUNCATE_EXISTING), digest))) {
+      if (bundleRequestFileList.getFiles().isEmpty()) {
+        // If file path is empty string which means we want to zip the whole bundle id folder
+        BundleJarUtil.addToArchive(uuidFile, zipOut);
```

Comment:
I think you should call `BundleJarUtil.addToArchive(uuidFile, true, zipOut)` instead to make the zip entry consistent with the `else` case, which the entry name contains the directory name as the prefix.

Reply:
I see